docs/DeveloperGuide.md

```diff
@@ -1,34 +1,383 @@
+functions sortByDate() and sortByPriority().
+
+
+### Storage component
+
+
+![StorageComponent](diagrams/StorageClass.png)
+
+
+At the initiating and closing stage of the program, the Storage component
+will read and write to the data file respectively. As shown in the diagram
+above, the data file is named as "data.json" and placed under a folder called
+"data". The Storage component also uses the Google Gson library to convert Task
+object to and from json format. For example, in the process of saving to file,
+each Task object is convert to json string and then writes to file via FileWriter. 
+
+
+### Common classes
+
+Common classes provide utility functions and string messages used by multiple components,
+and these classes are stored in the seedu.commons package.
 
 
 ## Product scope
+
 ### Target user profile
 
-{Describe the target user profile}
+    - Students in general
+    - Owns a personal laptop/PC
+    - Willing to use command line app for planning
+
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+    PlaNUS is a one stop location for students to be able to view all their 
+    commitments and assignments, and prioritize their work and commitment 
+    with a user-friendly text interface.
+
 
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|Forgetful student|add tasks to list|keep track of them and not forget it|
+|v1.0|Busy student|modify existing tasks|don't waste time deleting and adding it all over again|
+|v1.0|New user|see instructions|have quick access to the instruction w/o having to access the user guide|
+|v1.0|Student with many different tasks & due dates|set task priority|focus on more urgent tasks |
+|v1.0|Student with many different tasks & due dates|sort tasks by priority|clearly view of tasks urgency|
+|v1.0|Student with many tasks|search for tasks|find what i want quickly|
+|v1.0|Student with many tasks|delete tasks|old tasks don’t clutter up the display|
+|v1.0|Busy student|clear all tasks|don’t waste time deleting one by one|
+|v2.0|Busy student|View tasks in a weekly and monthly view|I know what i have due for that time period|
+
+
+## Use cases
+
+### Use case: Add task
+
+Main success scenario
+
+1. User adds a task
+2. PLANus adds task
+3. PLANus shows the task added message
+
+Use case ends. 
+
+Extensions  
+   - 1a. User adds task without any date  
+Date is set to the current date  
+   - 1b. User adds tasks without any priority  
+Priority is set to low  
+   - 1c. User adds task without start and end time  
+Time is set to empty  
+   - 1d. User inputs wrong details format  
+PLANus shows invalid command message  
+
```

Comment:
A sequence or object diagram can be drawn to better explain how each task/command operates.